src/rpdk/core/data_loaders.py

```diff
@@ -176,36 +182,45 @@ def load_resource_spec(resource_spec_file):  # pylint: disable=R # noqa: C901
                                 "maximum",
                                 "exclusiveMinimum",
                                 "exclusiveMaximum",
+                                "multipleOf",
                             },
                         ),
                         (
                             {"string"},
                             {
                                 "minLength",
                                 "maxLength",
+                                "pattern",
                             },
                         ),
                         (
                             {"object"},
                             {
                                 "minProperties",
                                 "maxProperties",
+                                "additionalProperties",
+                                "patternProperties",
                             },
                         ),
                         (
                             {"array"},
                             {
                                 "minItems",
                                 "maxItems",
+                                "additionalItems",
+                                "uniqueItems",
                             },
                         ),
                     ]:
                         if (
                             property_type in types
-                            and min_max_keywords - allowed_keywords & property_keywords
+                            and type_specific_keywords - allowed_keywords
+                            & property_keywords
```

Comment:
I am curious why do you use bitwise operator with logical ones? 
what takes priority - `type_specific_keywords - (allowed_keywords & property_keywords)` or `(type_specific_keywords - allowed_keywords) & property_keywords`? 

and why `property_keywords` is in it? coul you provide some examples?

Reply:
didn't mean to touch this part in this PR (formatted by `black`), but good question nonetheless :)

it's equivalent to `(type_specific_keywords - allowed_keywords) & property_keywords`

which is logically equivalent to `dis_allowed_keywords & property_keywords`, i.e. any intersection between keywords not allowed for that type and keywords present for that type